File: master-goods-android/core/model/src/main/java/com/zhihuiji/core/model/AuthModels.kt
package com.zhihuiji.core.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class LoginRequest(
    val phone: String,
    val password: String,
)

@Serializable
data class RegisterRequest(
    val phone: String,
    val password: String,
    @SerialName("verify_code") val verifyCode: String,
)

@Serializable
data class RefreshRequest(
    @SerialName("refresh_token") val refreshToken: String,
)

@Serializable
enum class VerificationType(@SerialName("type") val value: String) {
    @SerialName("register") REGISTER("register"),
    @SerialName("login") LOGIN("login"),
    @SerialName("reset_password") RESET_PASSWORD("reset_password"),
}

@Serializable
data class VerifyCodeRequest(
    val phone: String,
    val type: VerificationType,
)

@Serializable
data class VerifyCodeResponse(
    val success: Boolean,
    @SerialName("expire_seconds") val expireSeconds: Int,
)

@Serializable
data class AuthResult(
    @SerialName("user_id") val userId: Long,
    val token: String,
    @SerialName("refresh_token") val refreshToken: String,
    @SerialName("expires_in") val expiresIn: Int,
)

@Serializable
data class UserProfile(
    val id: Long,
    val phone: String,
    val nickname: String,
    val status: Int,
)

File: ios/ZhihuijiIOS/Core/Auth/AuthModels.swift
import Foundation

struct LoginRequest: Codable {
    let phone: String
    let password: String
}

struct RefreshRequest: Codable {
    let refreshToken: String
}

struct AuthPayload: Codable, Equatable {
    let userId: Int64
    let token: String
    let refreshToken: String?
    let expiresIn: Int?
}

struct UserProfile: Codable, Equatable {
    let id: Int64
    let phone: String
    let nickname: String
    let status: Int
}

struct APIEnvelope<T: Codable>: Codable {
    let code: Int?
    let message: String?
    let data: T
}
